### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by valakar**
## Evaluation criteria (each mark includes previous mark criteria)

Provide your reviewers with the link to the repo, client application and URLs to execute the **/import** path of the **import-service**

- [x]  **1** - **authorization-service** is added to the repo, has correct **basicAuthorizer** lambda and correct **serverless.yaml** file
- [x]  **3** - **import-service** serverless.yaml file has authorizer configuration for the **importProductsFile** lambda. Request to the **importProductsFile** lambda should work only with correct **authorization_token** being decoded and checked by **basicAuthorizer** lambda. Response should be in 403 HTTP status if access is denied for this user (invalid **authorization_token**) and in 401 HTTP status if Authorization header is not provided.
- [x]  **5** - update client application to send Authorization: Basic **authorization_token** header on import. Client should get **authorization_token** value from browser localStorage [https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage](https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage)**authorization_token** = localStorage.getItem('**authorization_token**')

## Additional (optional) tasks

- [x]  **+1** - Client application should display alerts for
the responses in 401 and 403 HTTP statuses. This behavior should be
added to the **nodejs-aws-fe-main/src/index.tsx** file

On page refresh app will set **wrongToken**, please change it (right - **dmFsYWthcjpURVNUX1BBU1NXT1JE**) or delete

[FE pull request](https://github.com/valakar/nodejs-aws-fe/pull/5)

[App](https://d1jpnfaozgam1v.cloudfront.net/)

### CODE REVIEW

**Comment by aleksei-bulgak-study** [comment]
Link to cloudfront distribution for other reviewers https://d1jpnfaozgam1v.cloudfront.net

**Comment by aleksei-bulgak-study** [comment]
please provide links for your application and cloudfront distribution
I have found this one in your previous PRs https://d1jpnfaozgam1v.cloudfront.net but looks like it does not contains latest UI changes

File: authorization-service/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "authorization-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using Typescript",
+  "main": "handler.js",
+  "scripts": {
+    "deploy": "npx sls deploy",
+    "deploy:importProductsFile": "npx sls deploy --function importProductsFile",
```
Comment (aleksei-bulgak-study) [inline authorization-service/package.json:8]: does this two scripts related to authorization-service?

**Comment by izzetbek** [comment]
401 error is returned with given token. There is no error handling on front-end (alerting error).

**Comment by valakar** [comment]
> please provide links for your application and cloudfront distribution
> I have found this one in your previous PRs https://d1jpnfaozgam1v.cloudfront.net but looks like it does not contains latest UI changes

Hi @aleksei-bulgak-study , thank you for your comment
Sorry, my bad, link added, [cloudfront](https://d1jpnfaozgam1v.cloudfront.net/) redeployed

File: authorization-service/package.json (hunk 1 above)
Comment (valakar) [reply authorization-service/package.json:8]: No, it was copypaste from import service
While AuthorizarionService consist of only one lambda, was using only deploy script and didn't notice this error, thanks

**Comment by valakar** [comment]
> 401 error is returned with given token. There is no error handling on front-end (alerting error).

@izzetbek App was redeploy, now should be properly handled error

image

**Comment by aleksei-bulgak-study** [comment]
When I use such token `YWxla3NlaS1pbnZhbGlkLXRva2VuOlRFU1RfUEFTU1dPUkQ=` then I can upload file which is strange for me. based on your code I should receive 403 status but I'm not sure why I receive success code

**Comment by valakar** [comment]
> When I use such token `YWxla3NlaS1pbnZhbGlkLXRva2VuOlRFU1RfUEFTU1dPUkQ=` then I can upload file which is strange for me. based on your code I should receive 403 status but I'm not sure why I receive success code

Hi @aleksei-bulgak-study , heh, it looks like issue was in **import-service**, I have used auto generated output ref "arn: `${cf:authorization-service-dev.BasicAuthorizerLambdaFunctionQualifiedArn}`"  and it looks like it store version during deploy of this function...(and in this case import-service was using previous version of basicAutorizer) 
Thank you for pointing on this! Already fixed

**Review by aleksei-bulgak-study — APPROVED** [review]

**Pull request merged**